golos.publication/golos.publication.cpp

```diff
@@ -152,106 +148,34 @@ void publication::upvote(account_name voter, account_name author, std::string pe
     require_auth(voter);
     limit_of_votes(voter);
 
-    tables::vote_table vote_table(_self, author);
-    tables::voters_table voters_table(_self, author);
+//    eosio_assert(weight > 0, "The weight sign can't be negative.");
 
     std::vector<structures::rshares> rshares;
 
     structures::rshares rshares_obj{0, 0, 0, 0};
 
     rshares.push_back(rshares_obj);
 
-    structures::post posttable_obj;
-
-    if (get_post(author, permlink, posttable_obj)) {
-        auto votetable_index = vote_table.get_index<N(postid)>();
-        auto votetable_obj = votetable_index.find(posttable_obj.id);
-        if (votetable_obj != votetable_index.end()) {
-            auto voterstable_index = voters_table.get_index<N(postid)>();
-            auto voterstable_obj = voterstable_index.find(posttable_obj.id);
-            while (voterstable_obj != voterstable_index.end() &&
-                   posttable_obj.id == voterstable_obj->post_id) {
-                eosio_assert(voter != voterstable_obj->voter,
-                             "You have already voted for this post.");
-                ++voterstable_obj;
-            }
-
-            voters_table.emplace(author, [&]( auto &item ) {
-                item.id = voters_table.available_primary_key();
-                item.post_id = posttable_obj.id;
-                item.voter = voter;
-            });
-
-            votetable_index.modify(votetable_obj, author, [&]( auto &item ) {
-               item.post_id = posttable_obj.id;
-               item.voter = voter;
-               item.percent = FIXED_CURATOR_PERCENT;
-               item.weight = weight;
-               item.time = now();
-               item.rshares = rshares;
-               ++item.count;
-            });
-        } else {
-            voters_table.emplace(author, [&]( auto &item ) {
-                item.id = voters_table.available_primary_key();
-                item.post_id = posttable_obj.id;
-                item.voter = voter;
-            });
-
-            vote_table.emplace(author, [&]( auto &item ) {
-               item.post_id = posttable_obj.id;
-               item.voter = voter;
-               item.percent = FIXED_CURATOR_PERCENT;
-               item.weight = weight;
-               item.time = now();
-               item.rshares = rshares;
-               ++item.count;
-            });
-        }
-    }
+    set_vote(voter, author, permlink, weight, rshares);
 }
 
 void publication::downvote(account_name voter, account_name author, std::string permlink, asset weight) {
     require_auth(voter);
 
-    tables::vote_table vote_table(_self, author);
-    tables::voters_table voters_table(_self, author);
+//    eosio_assert(weight < 0, "The weight sign can't be positive.");
```

Comment:
please, restore asserts after implementing #60. All weights must be positive and `downvote` must transform it to negative when call `set_vote`